### DISCUSSION TITLE
Configuration of Additional Algorithms

### DISCUSSION THREAD

**Discussion by johodges** [firetools/qgis2fds#62] [General]
There has been some offline discussion of how we want to organize additional algorithms within the qgis2fds project, but we wanted to move the discussion online. In the current develop branch, the WCS export and re-export terrain algorithms are configured as separate algorithms under the same qgis2fds provider plugin. Another approach would be to have a separate provider/plugin with additional tools available. This would allow us to only have the qgis2fds plugin installed through the official framework without the other tools we may add. My personal preference is to keep with the algorithms under the same provider for simplicity. Thoughts?

**Comment by rmcdermo** [comment]
I strongly favor keeping everything under one plugin.  For now, as Eric M is working his stuff, he is keeping that separated.  But I believe he is planning to add it to qgis2fds at some point.

**Comment by rmcdermo** [comment]
@emanuelegissi I suggest adding @ericvmueller as a collaborator.  Thanks

**Comment by emanuelegissi** [comment]
I also strongly favor keeping everything under one plugin.
The same qgis2fds plugin can contain several algorithms. I suggest
something like this:

NIST FDS
> Save remote layer     <-- both texture, DEM, and landuse data
> Export terrain to FDS case    <-- works only with local data
> Load FDS case results    <-- integrate Eric work

@Randy Ok, adding Eric

Il giorno gio 27 lug 2023 alle ore 13:42 Randy McDermott <
***@***.***> ha scritto:

> @emanuelegissi <https://github.com/emanuelegissi> I suggest adding
> @ericvmueller <https://github.com/ericvmueller> as a collaborator. Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/firetools/qgis2fds/discussions/62#discussioncomment-6564366>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXMVY5BH4PY4JM3F26EXSTXSJH3JANCNFSM6AAAAAA2Z57SME>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
"When old people used to wish each other peace, we didn't understand the
meaning of this word. Now we really understand." [Natalia Balasynovych,
mayor of Vasilkiv, Ukraine 2022]

**Comment by ericvmueller** [comment]
Thanks for adding me @emanuelegissi! 
I agree with that hierarchy of algorithms. I can start working on a branch that turns the [separate plugin](https://github.com/ericvmueller/fds_isochrones) into an additional 'load results' algorithm. Once it is re-structured in that way we can discuss the details of the algorithm and make sure it is consistent with the overall approach ...then merge into the firetools repo.

**Comment by emanuelegissi** [comment]
Great!
I just need a couple of days, as I am preparing the infrastructure for
multiple algorithms, and experimenting with object oriented parameters on
the dev branch

Il giorno gio 27 lug 2023 alle ore 14:39 Eric Mueller <
***@***.***> ha scritto:

> Thanks for adding me @emanuelegissi <https://github.com/emanuelegissi>!
> I agree with that hierarchy of algorithms. I can start working on a branch
> that turns the separate plugin
> <https://github.com/ericvmueller/fds_isochrones> into an additional 'load
> results' algorithm. Once it is re-structured in that way we can discuss the
> details of the algorithm and make sure it is consistent with the overall
> approach ...then merge into the firetools repo.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/firetools/qgis2fds/discussions/62#discussioncomment-6564898>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXMVYZJXLLO6C3Q5ZUKTUTXSJOQVANCNFSM6AAAAAA2Z57SME>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
"When old people used to wish each other peace, we didn't understand the
meaning of this word. Now we really understand." [Natalia Balasynovych,
mayor of Vasilkiv, Ukraine 2022]
